src/pages/Settings/global.tsx

```diff
@@ -0,0 +1,260 @@
+import { PlusOutlined } from '@ant-design/icons';
+import { Button, message, Tag, Tooltip, Popconfirm } from 'antd';
+import React, { useState, useRef } from 'react';
+import { useIntl, FormattedMessage } from 'umi';
+import type { IntlShape } from 'react-intl';
+import type { ProColumns, ActionType } from '@ant-design/pro-table';
+import ProTable from '@ant-design/pro-table';
+
+import {
+  settings,
+  createSettings,
+  updateSettings,
+  deleteSettings,
+} from '@/services/escola-lms/settings';
+
+import SettingsModalForm from './components/ModalForm';
+import { DeleteOutlined, EditOutlined } from '@ant-design/icons';
+
+const handleUpdate = async (intl: IntlShape, fields: API.Setting, id?: number) => {
+  const hide = message.loading(
+    intl.formatMessage({
+      id: 'loading',
+    }),
+  );
+  try {
+    await (id && id !== -1 ? updateSettings(id, { ...fields }) : createSettings({ ...fields }));
+    hide();
+    message.success(
+      intl.formatMessage({
+        id: 'success',
+      }),
+    );
+
+    return true;
+  } catch (error) {
+    hide();
+    message.error(
+      intl.formatMessage({
+        id: 'error',
+      }),
+    );
+    return false;
+  }
+};
+
+const handleRemove = async (intl: IntlShape, id: number) => {
+  const hide = message.loading(
+    intl.formatMessage({
+      id: 'loading',
+    }),
+  );
+  try {
+    await deleteSettings(id);
+    hide();
+    message.success(
+      intl.formatMessage({
+        id: 'success',
+      }),
+    );
+    return true;
+  } catch (error) {
+    hide();
+    message.error(
+      intl.formatMessage({
+        id: 'error',
+      }),
+    );
+    return false;
+  }
+};
+
+const staticLogo: API.Setting = {
+  id: -1,
+  key: 'logo',
+  group: 'global',
+  value: '',
+  public: true,
+  enumerable: true,
+  sort: 0,
+  type: 'image',
+  data: 'EscolaLMS',
+};
+
+const staticFrontURL: API.Setting = {
+  id: 0,
+  key: 'frontURL',
+  group: 'global',
+  value: '',
+  public: true,
+  enumerable: true,
+  sort: 0,
+  type: 'text',
+  data: 'EscolaLMS',
+};
+
+const TableList: React.FC = () => {
+  const [modalVisible, setModalVisible] = useState<number | false>(false);
```

Comment:
modal is either 

ID (existing record), initalData (new record with predefined fields), false (closed) 

```suggestion
  const [modalVisible, setModalVisible] = useState<number | Partial<API.Setting> | false>(false);
```